models.py

```diff
@@ -309,8 +318,9 @@ def update_wallet_address(cls, session, user):
                     if have_tx:
                         break
                     if tx["type"] == 4 and tx["assetId"] == app.config["ASSET_ID"]:
+                        amount_nzd = round(((tx['amount']/100) * float( 1 - rate)),2)
```

Comment:
dont convert amount_nzd to a floating point value (our schema says we store amount and amount_nzd as integers in the db)

take the equation here (https://github.com/zap-me/zapm_server/blob/3b1dc2c1b5a81d6c977f5bb2cc6d7cbda17c5cbf/app.py#L364) and move it into a function in utils.py and have both places use the same function to convert amount -> amount_nzd

Reply:
This calculation confuses me. What value are we putting on the merchant rate?
when you're in admin view the merchant rate by default is 5%.

By the equation its saying the if someone pays 5(zap), the merchant is receiving in nzd:

amount_receive = 5 / (1 + 0.05)
                           = 5 / 1.05
                           = 4.761904761904762‬

5% of 5zap is: 0.25

So the amount receive should 4.75

Or am I looking this all wrong.

Reply:
amount = zap in zapcents (integer)
amount_nzd = nzd in cents (integer)

5 NZD * (1 + 0.05) = 5.25 ZAP
5.25 ZAP / (1 + 0.05) = 5 NZD

Reply:
hmmm theres a discrepancy when i place the result side by side:

![image](https://user-images.githubusercontent.com/47799652/85088837-214e1e80-b235-11ea-8560-c9db5dd87ecd.png) and its a big one as the value grows.

Reply:
> amount = zap in zapcents (integer)
> amount_nzd = nzd in cents (integer)
> 
> 5 NZD * (1 + 0.05) = 5.25 ZAP
> 5.25 ZAP / (1 + 0.05) = 5 NZD

There's a discrepancy and the bigger the amount zap, the bigger the difference
![image](https://user-images.githubusercontent.com/47799652/85088982-7722c680-b235-11ea-8265-d4f92c0df019.png)

Reply:
push changes

Reply:
just pushed this.